File: Semestrul 4/MPP/Proiecte/java_networking/Networking/src/main/java/networking/utils/RpcConcurrentServer.java
package networking.utils;

import networking.rpcprotocol.ClientRpcReflectionWorker;
import services.IProjectServices;

import java.net.Socket;

public class RpcConcurrentServer extends AbsConcurrentServer{
    private IProjectServices Server;
    public RpcConcurrentServer(int port, IProjectServices Server1) {
        super(port);
        this.Server = Server1;
        System.out.println("RpcConcurrentServer");
    }

    @Override
    protected Thread createWorker(Socket client){
        ClientRpcReflectionWorker worker = new ClientRpcReflectionWorker(Server, client);

        Thread thread = new Thread(worker);
        return thread;
    }

    @Override
    public void stop(){
        System.out.println("Stopping services...");
    }
}

File: Semestrul 4/MPP/Proiecte/serverC#clientJava/c#/Utils/RpcConcurrentServer.cs
using servicesnet;

namespace networkingnet.Utils;

using System;
using System.Net.Sockets;
using System.Threading;
using networkingnet.RpcProtocol;


public class RpcConcurrentServer : AbsConcurrentServer
{
    private IProjectServices server;

    public RpcConcurrentServer(int port, IProjectServices transportServer) : base(port)
    {
        this.server = transportServer;
        Console.WriteLine("Transport - TransportRpcConcurrentServer");
    }

    protected override Thread CreateWorker(TcpClient client)
    {
        ClientJsonWorker worker = new ClientJsonWorker(server, client);
        Thread tw = new Thread(worker.Run);
        return tw;
    }

    public override void Stop()
    {
        Console.WriteLine("Stopping services ...");
        base.Stop();
    }
}